src/averageAges.js

```diff
@@ -55,7 +68,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const femalesOnly = people.filter(man => man.sex === 'f');
+  const mothersOnly = femalesOnly
+    .filter(mother => people
+      .find(children => mother.name === children.mother));
+  const motherOfSunsOnly = femalesOnly
+    .filter(mother => people
+      .find(children => {
+        return mother.name === children.mother && children.sex === 'm';
+      }));
+  const childrenOnly = people
+    .filter(man => mothersOnly
+      .find(mother => man.mother === mother.name));
+  const sonsOnly = people
+    .filter(man => mothersOnly
+      .find(mother => man.mother === mother.name && man.sex === 'm'));
+  const mothersList = onlyWithSon ? motherOfSunsOnly : mothersOnly;
+  const childrenList = onlyWithSon ? sonsOnly : childrenOnly;
+  const ageList = childrenList
+    .map(children => children.born - mothersList
+      .find(mother => mother.name === children.mother).born);
+  const sumAge = ageList.reduce((sum, age) => sum + age);
+
+  return sumAge / ageList.length;
```

Comment:
You are creating too many variables. In the future tasks - try to use chained methods more where possible.